### PR TITLE
[Kotlin Migration] NoteInfo

### PR DESCRIPTION
**Pull request by n1snt**
## Checklist

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented on your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Review by david-allison — CHANGES_REQUESTED** [review]
kotlinMigration.gradle does not support the `api` project.

A `kotlinMigration-api.gradle` will be required.

**Comment by n1snt** [comment]
Can I try creating a Gradle file for kotlinMigration for `api` project?
Where should I start?
And why wouldn't the `kotlinMigration.gradle` file work?
Would it need updating the paths in Source?

**Comment by david-allison** [comment]
kotlinMigration isn't included in the API 'build.grade': https://github.com/ankidroid/Anki-Android/blob/25473eeaff65c06c68d0d0e6fe79805da479437a/api/build.gradle

See: c16aae062ad470513e2d09640ef028f3a4a41c4b and https://github.com/ankidroid/Anki-Android/pull/11037 for how to make this file.

Ensure that you test it by building the first commit

**Comment by n1snt** [comment]
Now updated it to using `kotlinMigration-api.gradle`

**Comment by david-allison** [comment]
Blocked on resolution of https://github.com/ankidroid/Anki-Android/pull/11286#issuecomment-1123167003

**Comment by n1snt** [comment]
@david-allison I think this could be merged now.
Have a look at this & let me know if I should make any changes to it.

File: api/src/main/java/com/ichi2/anki/api/NoteInfo.kt [hunk 1]
```diff
@@ -14,76 +14,76 @@
...
 
 /**
  * Representation of the contents of a note in AnkiDroid.
  */
-public final class NoteInfo {
-    private final long mId;
-    private final String[] mFields;
-    private final Set<String> mTags;
+class NoteInfo {
+    private val id: Long
```
Comment (david-allison) [inline api/src/main/java/com/ichi2/anki/api/NoteInfo.kt:27]: Are you sure these weren't renamed by hand?

Comment (n1snt) [reply api/src/main/java/com/ichi2/anki/api/NoteInfo.kt:27]: These were named like this automatically during migration.

**Review by david-allison — APPROVED** [review]

**Pull request merged**